server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala

```diff
@@ -96,142 +100,306 @@ class AddressIndexRepository @Inject()(
+  def generateQueryAddressRequest(tokens: Map[String, String], originalInput: String): SearchDefinition = {
+
+    val saoQuery =
+      Seq(
+        tokens.get(Tokens.saoStartNumber).map(token =>
+          matchQuery(
+            field = "lpi.saoStartNumber",
+            value = token
+          ).boost(queryParams.subBuildingName.lpiSaoStartNumberBoost)),
+        tokens.get(Tokens.saoStartSuffix).map(token =>
+          matchQuery(
+            field = "lpi.saoStartSuffix",
+            value = token
+          ).boost(queryParams.subBuildingName.lpiSaoStartSuffixBoost)),
+        tokens.get(Tokens.saoEndNumber).map(token =>
+          matchQuery(
+            field = "lpi.saoEndNumber",
+            value = token
+          ).boost(queryParams.subBuildingName.lpiSaoEndNumberBoost)),
+        tokens.get(Tokens.saoEndSuffix).map(token =>
+          matchQuery(
+            field = "lpi.saoEndSuffix",
+            value = token
+          ).boost(queryParams.subBuildingName.lpiSaoEndSuffixBoost))
+      ).flatten
+
+    val subBuildingNameQuery =
+      Seq(
+        tokens.get(Tokens.subBuildingName).map(token =>
+          matchQuery(
+            field = "paf.subBuildingName",
+            value = token
+          ).boost(queryParams.subBuildingName.pafSubBuildingNameBoost)),
+        tokens.get(Tokens.subBuildingName).map(token =>
+          matchQuery(
+            field = "lpi.saoText",
+            value = token
+          ).boost(queryParams.subBuildingName.lpiSaoTextBoost))
+      ).flatten
+
+    val paoQuery =
+      Seq(
+        tokens.get(Tokens.paoStartNumber).map(token =>
+          matchQuery(
+            field = "lpi.paoStartNumber",
+            value = token
+          ).boost(queryParams.buildingName.lpiPaoStartNumberBoost)),
+        tokens.get(Tokens.paoStartSuffix).map(token =>
+          matchQuery(
+            field = "lpi.paoStartSuffix",
+            value = token
+          ).boost(queryParams.buildingName.lpiPaoStartSuffixBoost)),
+        tokens.get(Tokens.paoEndNumber).map(token =>
+          matchQuery(
+            field = "lpi.paoEndNumber",
+            value = token
+          ).boost(queryParams.buildingName.lpiPaoEndNumberBoost)),
+        tokens.get(Tokens.paoEndSuffix).map(token =>
+          matchQuery(
+            field = "lpi.paoEndSuffix",
+            value = token
+          ).boost(queryParams.buildingName.lpiPaoEndSuffixBoost))
+      ).flatten
+
+    val buildingNameQuery =
+      Seq(
+        tokens.get(Tokens.buildingName).map(token =>
+          matchQuery(
+            field = "paf.buildingName",
+            value = token
+          ).boost(queryParams.buildingName.pafBuildingNameBoost)),
+        tokens.get(Tokens.buildingName).map(token =>
+          matchQuery(
+            field = "lpi.paoText",
+            value = token
+          ).boost(queryParams.buildingName.lpiPaoTextBoost))
+      ).flatten.map(_.fuzziness("AUTO"))
```

Comment:
Do we know what "AUTO" does? Does it use scaling i.e. if len(string) < x then fuzziness = 1, else 2?